Log watch streaming does not work as expected

When a client reads log using LogWatch then the reading is blocked and never resumes if the client reads the data faster than the log file generates them.

The issue is probably hidden in InputStreamPumper line

while (keepReading && !Thread.currentThread().isInterrupted() && (length = in.read(buffer)) != -1)

when the client is at the end of the log then  in.read(buffer) probably returns -1 which causes the cycle ended and method terminated. Thus the reading from pod log is stopped.

When I have monitored the behaviour with debugger it is clearly visible that the executor thread process run() method of the pumper and suddenly it waits for new task that never arrives.

OK, the initial analysis was wrong - an exception is thrown in the run method
```
java.net.SocketTimeoutException: timeout
	at okio.Okio$3.newTimeoutException(Okio.java:207)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:261)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:215)
	at okio.RealBufferedSource.request(RealBufferedSource.java:71)
	at okio.RealBufferedSource.require(RealBufferedSource.java:64)
	at okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:270)
	at com.squareup.okhttp.internal.http.Http1xStream$ChunkedSource.readChunkSize(Http1xStream.java:454)
	at com.squareup.okhttp.internal.http.Http1xStream$ChunkedSource.read(Http1xStream.java:435)
	at okio.RealBufferedSource$1.read(RealBufferedSource.java:371)
	at java.io.InputStream.read(InputStream.java:101)
	at io.fabric8.kubernetes.client.utils.InputStreamPumper.run(InputStreamPumper.java:48)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketException: Socket closed
	at java.net.SocketInputStream.read(SocketInputStream.java:203)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	at okio.Okio$2.read(Okio.java:139)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:211)
	... 13 more
```

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java
@@ -154,7 +154,9 @@ public class PodOperationsImpl extends HasMetadataOperation<Pod, PodList, Doneab
             URL url = new URL(URLUtils.join(getResourceUrl().toString(), getLogParameters() + "&follow=true"));
             Request request = new Request.Builder().url(url).get().build();
             final LogWatchCallback callback = new LogWatchCallback(out);
-            client.newCall(request).enqueue(callback);
+            OkHttpClient clone = client.clone();
+            clone.setReadTimeout(0, TimeUnit.MILLISECONDS);
+            clone.newCall(request).enqueue(callback);
             callback.waitUntilReady();
             return callback;
         } catch (Throwable t) {

